include/zephyr/min_heap.h

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2025 Aerlync Labs Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_MIN_HEAP_H_
+#define ZEPHYR_INCLUDE_MIN_HEAP_H_
+
+#include <zephyr/sys/util.h>
+#include <zephyr/kernel.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief min_heap
+ * @defgroup min_heap Min-heap service
+ * @ingroup os_services
+ * @{
+ */
+
+/**
+ * @brief Invasive node used inside user structs for min-heap support.
+ */
+struct min_heap_node {
+	/** Key for comparison (optional) */
+	int key;
+	/** Index within heap array */
+	size_t index;
+};
+
+/**
+ * @brief Comparator function type for min-heap ordering.
+ *
+ * This function compares two heap nodes to establish their relative order.
+ * It must be implemented by the user and provided at min-heap
+ * initialization.
+ *
+ * @param a Pointer to the first node.
+ * @param b Pointer to the second node.
+ *
+ * @return Negative value if @p a is less than @p b,
+ *         positive value if @p a is greater than @p b,
+ *         zero if they are equal.
+ */
+typedef int (*min_heap_cmp_fn)(const struct min_heap_node *a,
+			       const struct min_heap_node *b);
+
+/**
+ * @brief min-heap data structure with user-provided comparator.
+ */
+struct min_heap {
+	/** Pointer to preallocated node pointer array */
+	struct min_heap_node **nodes;
+	/** Maximum number of elements */
+	size_t capacity;
+	/** Current elements count */
+	size_t size;
+	/** Spinlock for thread safety */
+	struct k_spinlock lock;
```

Comment:
Design point:  traditionally in Zephyr, data structure libraries (c.f. slist, dlist, rbtree, even sys_heap) are unsynchronized, allowing the embedding environment to decide on how best to do that.  Lots of these things end up being used inside of more complicated locking already and don't need a private lock.

Suggest removing this.  It's not wrong to do it here, but it's definitely non-standard.  And I don't see that heapsort/priority-queues are really so special that they need their own synchronization management.